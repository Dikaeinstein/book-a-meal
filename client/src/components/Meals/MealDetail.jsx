import React from 'react';
import PropTypes from 'prop-types';

const MealDetail = ({ meal, closeMealDetail }) => {
  const handleCloseMealDetail = () => closeMealDetail();

  const mealDetailItemStyle = {
    padding: '0 1rem 1rem 1rem',
  };

  return (
    <div className="flex-container">
      <div style={{ ...mealDetailItemStyle, flex: '60%' }}>
        <img src={meal.imageUrl} alt={meal.name} />
      </div>
      <div style={{ ...mealDetailItemStyle, flex: '40%', color: '#000' }}>
        <h4
          className="text-capitalize font-weight-bold"
          style={{ margin: '.875rem 0' }}
        >
          {meal.name}
        </h4>
        <p
          className="font-weight-bold"
          style={{ margin: '.5rem 0' }}
        >
          Description:
        </p>
        <p style={{ fontSize: '.875rem', margin: '.5rem 0' }}>
          {meal.description}
        </p>
        <p style={{ margin: '.875rem 0' }}>
          <span className="font-weight-bold">
            Caterer:{' '}
          </span>
          <span>{meal.user.name}</span>
        </p>
        <div
          onClick={handleCloseMealDetail}
          onKeyPress={handleCloseMealDetail}
          className="text-dark font-weight-bold"
          style={{ cursor: 'pointer', fontSize: '.85rem' }}
          role="button"
          tabIndex="0"
        >
          Back to meals
        </div>
      </div>
    </div>
  );
};

MealDetail.propTypes = {
  meal: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.object,
  ])).isRequired,
  closeMealDetail: PropTypes.func.isRequired,
};

export default MealDetail;
