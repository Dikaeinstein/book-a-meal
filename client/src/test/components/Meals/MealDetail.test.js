import React from 'react';
import { shallow } from 'enzyme';
import MealDetail from '../../../components/Meals/MealDetail';

describe('<MealDetail />', () => {
  const props = {
    closeMealDetail: jest.fn(),
    meal: {
      name: 'Test meal',
      description: 'This is a test meal',
      user: { name: 'Test Dika' },
    },
  };
  it('should call handleCloseMealDetail when backToMeals is clicked', () => {
    const wrapper = shallow(<MealDetail {...props} />);
    const closeMealDetail = wrapper.find('div[role="button"]');
    closeMealDetail.simulate('click');
    expect(props.closeMealDetail).toHaveBeenCalled();
  });
  it('should match snapshot', () => {
    const tree = shallow(<MealDetail {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
