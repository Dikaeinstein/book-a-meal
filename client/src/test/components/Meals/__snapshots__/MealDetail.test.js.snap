// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealDetail /> should match snapshot 1`] = `
<div
  className="flex-container"
>
  <div
    style={
      Object {
        "flex": "60%",
        "padding": "0 1rem 1rem 1rem",
      }
    }
  >
    <img
      alt="Test meal"
    />
  </div>
  <div
    style={
      Object {
        "color": "#000",
        "flex": "40%",
        "padding": "0 1rem 1rem 1rem",
      }
    }
  >
    <h4
      className="text-capitalize font-weight-bold"
      style={
        Object {
          "margin": ".875rem 0",
        }
      }
    >
      Test meal
    </h4>
    <p
      className="font-weight-bold"
      style={
        Object {
          "margin": ".5rem 0",
        }
      }
    >
      Description:
    </p>
    <p
      style={
        Object {
          "fontSize": ".875rem",
          "margin": ".5rem 0",
        }
      }
    >
      This is a test meal
    </p>
    <p
      style={
        Object {
          "margin": ".875rem 0",
        }
      }
    >
      <span
        className="font-weight-bold"
      >
        Caterer:
         
      </span>
      <span>
        Test Dika
      </span>
    </p>
    <div
      className="text-dark font-weight-bold"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": ".85rem",
        }
      }
      tabIndex="0"
    >
      Back to meals
    </div>
  </div>
</div>
`;
